front/src/applications/stdcmV2/components/StdcmConsist.tsx

```diff
@@ -129,24 +140,75 @@ const StdcmConsist = ({ setCurrentSimulationInputs, disabled = false }: StdcmCon
           id="tonnage"
           label={t('consist.tonnage')}
           trailingContent="t"
-          inputFieldWrapperClassname="weight"
+          type="number"
+          min={0}
+          value={totalMass ?? ''}
+          onChange={onTotalMassChange}
         />
         <Input
           id="length"
           label={t('consist.length')}
           trailingContent="m"
-          inputFieldWrapperClassname="length"
+          type="number"
+          min={0}
+          value={totalLength ?? ''}
+          onChange={onTotalLengthChange}
+        />
+      </div>
+      <div className="stdcm-v2-consist__properties">
+        <SpeedLimitByTagSelector
+          disabled={disabled}
+          selectedSpeedLimitByTag={speedLimitByTag}
+          speedLimitsByTags={speedLimitsByTags}
+          dispatchUpdateSpeedLimitByTag={dispatchUpdateSpeedLimitByTag}
+        />
+        <Input
+          id="maxSpeed"
+          label={t('consist.maxSpeed')}
+          trailingContent="km/s"
+          type="number"
+          min={0}
+          value={maxSpeed ?? ''}
+          onChange={onMaxSpeedChange}
         />
       </div>
-      <SpeedLimitByTagSelector
-        disabled={disabled}
-        selectedSpeedLimitByTag={speedLimitByTag}
-        speedLimitsByTags={speedLimitsByTags}
-        dispatchUpdateSpeedLimitByTag={dispatchUpdateSpeedLimitByTag}
-        className="speed-limit-by-tag-selector"
-      />
     </StdcmCard>
   );
 };
 
+function useStdcmConsist() {
```

Comment:
Thank you for the change. Could you move the hook in a `useStdcmConsist.ts` file in the "hooks" folder ? (in stdcmV2 folder)